### ISSUE TITLE
Consistent header layout across all sub-domains

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe.org#105]
I would like to have a somewhat consistent header design across all our sub-domains. What I imagine is that the top-left looks like this:

HaxeLogo >> SubDomainLogo

The >> should indicate that we are one level "below" main haxe.org. I'm not sure how exactly that should look but we can figure something out.

Then the HaxeLogo can always link back to haxe.org while the SubDomainLogo links to lib.haxe.org, api.haxe.org, try.haxe.org and whatever else we might have. This is good for usability because you have a consistent way of getting back to the main page of both haxe.org and the current sub-domain.

**Comment by jasononeil** [comment]
:+1:

**Comment by ncannasse** [comment]
fine for me, although we maybe don't want a subdomain "logo" (two images side by side) but simply the "text" for it , like   (Haxe Log) API  or  (Haxe Log) Try

**Comment by markknol** [comment]
if we use the logo, then a orange or gray rectangle (from top to bottom in header bar) with the subsection word in it, that can look already great. Maybe with a subtle detail in it, if I can find some time I might want to design this.

**Comment by markknol** [comment]
What about this?
![haxe-site-header](https://cloud.githubusercontent.com/assets/576184/6651836/fdde9324-ca56-11e4-8d85-64d19fcbb0b2.gif)
(click to preview fullscreen)

* The subsection site get their own logo part with subtle arrow in it as suggested
* Search is everywhere the same, not sure if/how this is possible
* User is complete optional, but in case we ever want to store snippets per user, manage haxelib :wink:

**Comment by Simn** [comment]
That's almost exactly what I had in mind. Unfortunately that usually means that Nicolas is not going to like it.

**Comment by ncannasse** [comment]
I had the same thing in mind as well :) (miracle does exist)

**Comment by markknol** [comment]
Woaw 
![agree](http://assets.diylol.com/hfs/ad1/99f/0bd/resized/all-the-things-meme-generator-agree-to-all-the-things-2d75b5.jpg)

Now we only need to build it.

**Comment by jasononeil** [comment]
Looks great :+1:

**Comment by Simn** [comment]
@markknol: Can we get this sorted out soon?

**Comment by markknol** [comment]
@jasononeil @clemos  If I create a separate (bootstrap) html-page with the headers and provide the markup/css, could you implement it? Would that work?

**Comment by clemos** [comment]
Looks good :)

**Comment by markknol** [comment]
Just started on the navigation tonight, see progress:  http://projects.stroep.nl/haxe/navigation/
For now, I have only worked on the haxe.org navigation, still have to do the sub site thingies. 

I have to figure out how to do this with some magical CSS or SVG trick  :smile: 
![image](https://cloud.githubusercontent.com/assets/576184/8095301/43c791c2-0fcd-11e5-84da-7696a6283608.png)

**Comment by ncannasse** [comment]
@markknol can we keep the main menu casing as it's on haxe.org (not with UPPERCASE TITLES) thanks :)

**Comment by markknol** [comment]
I have updated the navigation test page.

**These are my thoughts**  
* The *haxe.org* and *api.haxe.org* share the same menu completely. 
* The *try.haxe.org* and *lib.haxe.org* have own menus, but share the "Learn Haxe" and "Connect" menus.
* I removed the uppercase :cry: 
* There are some references to pages that needs to be created (in Haxe Foundation submenu).

**Inplementation details**
* I used a single page with plain bootstrap menu. When we implement this thing, we can remove all menu specific stuff from current stylesheets. 
* I created a separate `haxe-nav.css` file which contains the styling for all navigations. 

**TODO**
* I have no idea how to deal with that search across all sites
* I still have to check the small-screen version

**tl;dr** http://projects.stroep.nl/haxe/navigation/
> use the bottom test navigation to fake a switch between the sites

**Comment by ncannasse** [comment]
Nice !
looking forward to have this integrated when @jasononeil is back

**Comment by ncannasse** [comment]
Just one nitpicking : do we really need to have a Google+ link ? (cc @Merelleya) it feels so... 2011

**Comment by markknol** [comment]
We can wait, maybe one day Google+ is as retro as IRC. Retro is always hot.

**Comment by Simn** [comment]
Looks really nice!

My only concern is that once you leave a sub-domain by clicking the Haxe logo it's not obvious how to get back to where you were. I still think it would be nice to have the sub-domain "button" as a dropdown for that, but then we don't have an obvious way for getting back to the sub-domain entry page itself.

**Comment by Merelleya** [comment]
Let me just repeat this, Mark, You are a start! Really. I love the new Menu. If you have the time, I would like to go through some details with you (I Am unsure whether I want the social pages and the support things together, mainly...I will meditate about this).

@ncannasse I think we underestimate and under-use google+. It never really got traction in Europe, but is more popular in the US. It is mainly used for business reasons, though (in connection with the google places/local business entry which links to the g+ page automatically). Especially if the Foundation were to ever have a physical address and get a google local entry, we will want to hold on to it.

We do need to give it some love, though, and decide how exactly we want to use it.

**Comment by clemos** [comment]
I'd be very happy to help on this, at least by integrating the menu to try.haxe.org.

Also, I agree with @Simn, there should be a more obvious way to navigate among the sites.
Maybe instead of "Learn Haxe", the "main" menu should be "About Haxe" or "What is Haxe" (I'm not too good in wordings), 
and contain a list of links to these "main sites" so it's obvious that they are the "first ring".
It could be something bigger than a small dropdown, something more like a full-width expand header menu with one column for each site, so we can also put a couple of sub-sections in it (Manual, About, maybe Download ?)
If it's not clear to you guys, I can definitely take @markknol 's stuff and hack it to show what I mean.

As a side note, I think it could be a good idea to store the whole menu / webring tree structure to some JSON / API, so it's easier to maintain.
I can also try and provide a basic structure / architecture if you think it's a good idea.

Finally (!), we (SilexLabs) are in the process of reorganizing the WWX archives as well,
so I thought it could be a good idea to include WWX documentation to this whole "webring" stuff.
I believe it would be a win-win situation.
What do you think ?

**Comment by markknol** [comment]
@clemos sounds like an idea. I personally was thinking that the sublogo could also be a dropdown, maybe that is the same "first ring" idea you have?

Please take my test navigation project and take it to a next level. It would be nice if we first have it complete it before it gets integrated? 
I created a temporary repo for it https://github.com/markknol/haxe-navigation

**Comment by Simn** [comment]
Does anyone feel responsible for getting this live?

**Comment by clemos** [comment]
I've been working a bit on it back then, 
but didn't manage to do anything convincing, 
and then, quite quickly, I forgot...
So, sorry, but I can't take the lead on this :(

**Comment by Merelleya** [comment]
@markknol ping

**Comment by markknol** [comment]
@Merelleya Hey, why are you pinging me? :smile:  I am working on it already for the try.haxe site, and I also want to do it for the haxe.org site, but I have troubles running that one locally because of some dependencies.

**Comment by Merelleya** [comment]
That's all I Wanted to know. I kind of lost track on who is taking care of this for what and I was looking at haxedevelop.org and noticed on the main haxe.org site there is no way to get back to the subdomains and so on.

**Comment by markknol** [comment]
If all have the new nav, that would be solved, right?

**Comment by Simn** [comment]
That's the idea!

**Comment by ibilon** [comment]
Only haxelib doesn't have the new layout, can be followed there https://github.com/HaxeFoundation/haxelib/pull/327

Will look at it after the haxe.org changes.
